spec/features/auto_deploy_spec.rb

```diff
@@ -413,11 +413,36 @@
 
   scenario 'receiving a `push` event for a deleted branch' do
     expect(github).to_not receive(:create_deployment)
-    push_event 'secret', deleted: true, head_commit: nil
+    push_event 'secret', deleted: true, head_commit: { id: "595ebd4ca061c4671ba89202aaf19c896f216635" }
   end
 
   scenario 'receiving a `push` event for a fork' do
     expect(github).to_not receive(:create_deployment)
     push_event 'secret', repository: { fork: true }
   end
+
+  scenario 'receiving a `push` event for an already auto deployed sha' do
+    config = <<-YAML.strip_heredoc
+    environments:
+      production:
+        continuous_delivery:
+          ref: refs/heads/master
+    YAML
+
+    allow(github).to receive(:contents).and_return(config)
+
+    expect(github).to receive(:create_deployment).with \
+      users(:david),
+      DeploymentRequest.new(
+        repository: 'baxterthehacker/public-repo',
+        ref: '595ebd4ca061c4671ba89202aaf19c896f216635',
+        environment: 'production',
+        force: true
+      )
+    push_event 'secret', sender: { id: github_accounts(:david).id }, head_commit: { id: "595ebd4ca061c4671ba89202aaf19c896f216635" }
```

Comment:
Is there a better way to ensure that this sha is already committed to the `AutoDeployment` model?